rosbag2_transport/include/rosbag2_transport/player.hpp

```diff
@@ -253,22 +253,21 @@ class Player : public rclcpp::Node
   void prepare_publishers();
   bool publish_message(rosbag2_storage::SerializedBagMessageSharedPtr message);
   static callback_handle_t get_new_on_play_msg_callback_handle();
-  static constexpr double read_ahead_lower_bound_percentage_ = 0.9;
-  static const std::chrono::milliseconds queue_read_wait_period_;
-  std::atomic_bool cancel_wait_for_next_message_{false};
-
-  std::mutex reader_mutex_;
-  std::unique_ptr<rosbag2_cpp::Reader> reader_ RCPPUTILS_TSA_GUARDED_BY(reader_mutex_);
-
   void add_key_callback(
     KeyboardHandler::KeyCode key,
     const std::function<void()> & cb,
     const std::string & op_name);
   void add_keyboard_callbacks();
-
   void create_control_services();
-
   void configure_play_until_timestamp();
+  bool shall_stop_at_timestamp(const rcutils_time_point_value_t & msg_timestamp) const;
```

Comment:
Does it make sense to change this method to be: `shall_continue_at_timestamp` so as to avoid negating in every usage occurrence?

Reply:
@agalbachicar Nop. I don't think so.  
I deliberately named it `shall_stop_at_timestamp` because it exactly reflects logic which is supposed to check for `play_until` and `play_for_duration` features.  
Don't worry about negation in each while loop it will not affect performance.